tests/run/lst/Lst.scala

```diff
@@ -252,9 +333,34 @@ class Lst[+T](val elems: Any) extends AnyVal {
       val newElems = new Arr(len1 + len2)
       this.copyToArray(newElems, 0)
       that.copyToArray(newElems, len1)
-      new Lst[U](newElems)
+      multi[U](newElems)
     }
 
+  def :: [U >: T](x: U): Lst[U] = elems match {
+    case null => Lst(x)
+    case elems: Arr =>
+      val newElems = new Arr(1 + elems.length)
+      newElems(0) = x
+      copyToArray(newElems, 1)
+      multi[U](newElems)
+    case elem: T @unchecked =>
+      Lst(x, elem)
+  }
+
+  def +: [U >: T](x: U): Lst[U] = elems match {
+    case null => Lst(x)
+    case elems: Arr =>
+      val newElems = new Arr(elems.length + 1)
+      copyToArray(newElems, 0)
+      newElems(elems.length) = x
+      multi[U](newElems)
+    case elem: T @unchecked =>
+      Lst(elem, x)
```

Comment:
In scala2 `head +: tail` has the same behaviour as `head :: tail`. This implementation looks like `init :+ last`.
I suspect this method should be named `:+`.